Jenkinsfile

```diff
@@ -13,38 +13,43 @@ elifePipeline {
             }
 
             stage 'Project tests', {
-                // ensure a clean starting state
-                sh "docker-compose down -v"
-                sh "docker-compose run ci ./project_tests.sh"
-                // preserve environment to allow investigation if build fails, clean up otherwise
-                sh "docker-compose down -v"
+                withCommitStatus({
+                    // ensure a clean starting state
+                    sh "docker-compose down -v"
+
+                    sh "docker-compose run ci ./project_tests.sh"
+                    // it is not yet possible to retrieve a JUnit XML log to archive as a test artifact:
+                    // - the `xunit` formatter mangles the XML outputting also debug statements between tags
+                    // - the `xunit-file` formatter, which is an external plugin, doesn't seem to work with gulp-mocha-phantomjs
+                }, 'project-tests', commit)
             }
-        },
-        'containers--medium'
-    )
 
-    stage 'Smoke tests', {
-        lock('pattern-library--ci') {
-            builderDeployRevision 'pattern-library--ci', commit
-            builderSmokeTests 'pattern-library--ci', '/srv/pattern-library'
-            // it is not yet possible to retrieve a JUnit XML log to archive as a test artifact:
-            // - the `xunit` formatter mangles the XML outputting also debug statements between tags
-            // - the `xunit-file` formatter, which is an external plugin, doesn't seem to work with gulp-mocha-phantomjs
+            stage 'Smoke tests', {
+                withCommitStatus({
```

Comment:
Doesn't seem to cause the links to go to the right step, but not that important.

Reply:
The manually generated commit statuses always to go [`env.RUN_DISPLAY_URL`](https://github.com/elifesciences/elife-jenkins-workflow-libs/blob/master/vars/withCommitStatus.groovy), but in case of failures Jenkins may open the failed step.